app/controllers/internal/events_controller.rb

```diff
@@ -0,0 +1,140 @@
+module Internal
+  class EventsController < ::InternalController
+    before_action :load_buildings, only: %i[new edit create update]
+    layout "internal"
+
+    def index
+      @no_results = load_pending_events.blank?
+    end
+
+    def show
+      @event = GetIntoTeachingApiClient::TeachingEventsApi.new.get_teaching_event(params[:id])
+      @page_title = @event.name
+      if @event.status_id != GetIntoTeachingApiClient::Constants::EVENT_STATUS["Pending"]
+        raise ActionController::RoutingError, "Not Found"
+      end
+    end
+
+    def new
+      @event = Event.new
+      @event.building = EventBuilding.new
+      @event.building.fieldset = "existing"
+    end
+
+    def final_submit
+      event = GetIntoTeachingApiClient::TeachingEventsApi.new.get_teaching_event(params[:format])
+      @event = transform_event(event)
+      @event.start_at = Time.zone.parse(@event.start_at.to_s)
+      @event.end_at = Time.zone.parse(@event.end_at.to_s)
+      if @event.approve
+        redirect_to internal_events_path(success: true)
+      else
+        render action: :new
+      end
+    end
+
+    def create
+      @event = Event.new(event_params)
+      @event.building = format_building(event_params["building"])
+      if @event.submit_pending
+        redirect_to internal_events_path(success: :pending)
+      else
+        render action: :new
+      end
+    end
+
+    def edit
+      event = GetIntoTeachingApiClient::TeachingEventsApi.new.get_teaching_event(params[:id])
+      @event = transform_event(event)
+      if @event.building.nil?
+        @event.building = EventBuilding.new(fieldset: "none")
```

Comment:
I wonder if the `fieldset` attribute should live in the `event` rather than the `building`? having `EventBuilding.new(fieldset: "none")` feels contradictory